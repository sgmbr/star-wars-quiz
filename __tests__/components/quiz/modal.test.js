import React from 'react'
import renderer from 'react-test-renderer'
import Modal from '../../../src/components/quiz/modal'

describe('Modal component', () => {
  it('renders modal', () => {
    const onClick = () => {}
    const component = renderer.create(
      <Modal
        modalTitle='Correct Answer'
        answer='Luke Skywalker'
        onClick={onClick}
      />
    )

    expect(component).toMatchSnapshot()
  })

  it('throws error with wrong prop type', () => {
    const onClick = () => {}
    const component = renderer.create(
      <Modal
        modalTitle={onClick}
      />
    )

    expect(component).toMatchSnapshot()
  })
})
