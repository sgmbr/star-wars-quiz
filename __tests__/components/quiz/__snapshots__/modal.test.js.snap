// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders modal 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="checkAnswer"
  className="modal fade"
  id="checkAnswer"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="checkAnswerLabel"
        >
          Correct Answer
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        The answer is "
        Luke Skywalker
        "
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-primary"
          data-dismiss="modal"
          onClick={[Function]}
          type="button"
        >
          Next Question
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal component throws error with wrong prop type 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="checkAnswer"
  className="modal fade"
  id="checkAnswer"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="checkAnswerLabel"
        />
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        The answer is "
        "
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-primary"
          data-dismiss="modal"
          onClick={undefined}
          type="button"
        >
          Next Question
        </button>
      </div>
    </div>
  </div>
</div>
`;
